astropy/coordinates/angles.py

```diff
@@ -386,22 +388,45 @@ def _wrap_at(self, wrap_angle):
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
         a360 = u.degree.to(self.unit, 360.0)
-        wrap_angle = wrap_angle.to_value(self.unit)
-        wrap_angle_floor = wrap_angle - a360
-        self_angle = self.view(np.ndarray)
-        # Do the wrapping, but only if any angles need to be wrapped
-        #
-        # This invalid catch block is needed both for the floor division
-        # and for the comparisons later on (latter not really needed
-        # any more for >= 1.19 (NUMPY_LT_1_19), but former is).
-        with np.errstate(invalid='ignore'):
-            wraps = (self_angle - wrap_angle_floor) // a360
-            valid = np.isfinite(wraps) & (wraps != 0)
-            if np.any(valid):
-                self_angle -= wraps * a360
-                # Rounding errors can cause problems.
-                self_angle[self_angle >= wrap_angle] -= a360
-                self_angle[self_angle < wrap_angle_floor] += a360
+        wrap_angle = wrap_angle.to_value(self.unit).astype(self.dtype)
+
+        view = self.view(np.ndarray)
+
+        # to support the case of non-native byteorder (not supported by cython)
+        # we have to switch to native. We later switch back to keep the input byteorder
+        # this was still faster in benchmarks than the previous numpy solution
+        switch_byteorder = False
+        if view.dtype.byteorder != "=":
+            data = view.byteswap().newbyteorder()
+            switch_byteorder = True
+        else:
+            data = view
+
+        # if the underlying array is read-only, we only check if it wraps
+        # and raise an error if any value wraps
+        if not view.flags.writeable:
+            if _needs_wrapping(data, wrap_angle, a360):
+                raise ValueError("Angle is not writeable but contains values outside wrapping range")
+            else:
+                return
+
+        if data.ndim == 0:
+            _wrap_at(data[np.newaxis], wrap_angle, a360)
+        elif data.ndim == 1:
+            _wrap_at(data, wrap_angle, a360)
+        else:
+            iter = np.nditer(
+                data,
+                op_flags=['readwrite'],
+                flags=["external_loop"],
+            )
+            for chunk in iter:
+                _wrap_at(chunk, wrap_angle, a360)
+
+        # swap back and mutate self
+        if switch_byteorder:
+            data = data.byteswap().newbyteorder()
+            view[...] = data
```

Comment:
I don't think Numpy preserves the byte order when doing operations on an array ?

Reply:
Could you clarify? I don't understand want you mean with this

Reply:
Sorry I missed your comment. What I mean is that Numpy doesn't preserve the byte order:
```
In [2]: (np.arange(5, dtype='>f4') + 1).dtype
Out[2]: dtype('float32')
```
I don't think it's useful to switch back to the input array's endianness.

Reply:
But we mutate the input buffer, so we need to push back values in the correct endianess, or not?